packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -191,4 +191,44 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
     const {count} = await this.getRawDb().one(`SELECT COUNT(*) FROM "Users"`);
     return count;
   }
+  
+  async getRandomActiveUser(): Promise<DbUser> {
+    return this.one(`
+      SELECT u.*
```

Comment:
Part of me feels like we should have a warning on these functions that they're going to be _real_ slow. Like, it's fine for this page, but I don't want someone to come along later and use them in a performance critical code path. That might just be over the top though.

Reply:
Seemed acceptable (this is prod):

```
 Limit  (cost=61417.39..61417.39 rows=1 width=4197) (actual time=428.624..428.687 rows=1 loops=1)
   ->  Sort  (cost=61417.39..61419.89 rows=1001 width=4197) (actual time=428.623..428.685 rows=1 loops=1)
         Sort Key: (random())
         Sort Method: top-N heapsort  Memory: 29kB
         ->  Nested Loop  (cost=50486.49..61412.39 rows=1001 width=4197) (actual time=364.414..417.635 rows=4689 loops=1)
               ->  Finalize GroupAggregate  (cost=50486.08..52053.00 rows=2002 width=26) (actual time=364.338..392.588 rows=4745 loops=1)
                     Group Key: "ReadStatuses"."userId"
                     Filter: (max("ReadStatuses"."lastUpdated") > (now() - '1 mon'::interval))
                     Rows Removed by Filter: 22599
                     ->  Gather Merge  (cost=50486.08..51887.81 rows=12014 width=26) (actual time=364.309..381.907 rows=52278 loops=1)
                           Workers Planned: 2
                           Workers Launched: 2
                           ->  Sort  (cost=49486.06..49501.07 rows=6007 width=26) (actual time=355.954..358.205 rows=17426 loops=3)
                                 Sort Key: "ReadStatuses"."userId"
                                 Sort Method: quicksort  Memory: 2325kB
                                 Worker 0:  Sort Method: quicksort  Memory: 2052kB
                                 Worker 1:  Sort Method: quicksort  Memory: 1629kB
                                 ->  Partial HashAggregate  (cost=49048.97..49109.04 rows=6007 width=26) (actual time=318.012..321.160 rows=17426 loops=3)
                                       Group Key: "ReadStatuses"."userId"
                                       Batches: 1  Memory Usage: 3089kB
                                       Worker 0:  Batches: 1  Memory Usage: 2833kB
                                       Worker 1:  Batches: 1  Memory Usage: 2577kB
                                       ->  Parallel Seq Scan on "ReadStatuses"  (cost=0.00..44019.98 rows=1005798 width=26) (actual time=0.005..82.533 rows=802164 loops=3)
               ->  Index Scan using "Users_pkey" on "Users" u  (cost=0.41..4.66 rows=1 width=4189) (actual time=0.004..0.004 rows=1 loops=4745)
                     Index Cond: ((_id)::text = ("ReadStatuses"."userId")::text)
                     Filter: (NOT COALESCE(deleted, false))
                     Rows Removed by Filter: 0
 Planning Time: 2.938 ms
 Execution Time: 429.211 ms
(29 rows)
```